stdlib/Random/src/Random.jl

```diff
@@ -20,13 +25,15 @@ export srand,
        randperm, randperm!,
        randcycle, randcycle!,
        AbstractRNG, MersenneTwister, RandomDevice,
-       GLOBAL_RNG, randjump
+       defaultRNG, randjump
 
 
 ## general definitions
 
 abstract type AbstractRNG end
 
+defaultRNG() = GLOBAL_RNG
```

Comment:
This shouldn't be camel-cased; it violates the style used everywhere else in Base.

Reply:
Also if this is just reading from a global const, why not just export the const rather than hiding it behind a function call?

Reply:
> This shouldn't be camel-cased;

I didn't really sea this as CamelCased, more as "initials get capitalized"; what would be prefered here? `default_rng` ? (`defaultrng` is un-readable).

> why not just export the const rather than hiding it behind a function call

If I remember correctly, it was needed at one point in this PR, because `sprand` had `GLOBAL_RNG` in its signature, but was in base and couldn't reference a not-yet created global object. Now that `SparseArrays` is in stdlib, it should be fine to use `GLOBAL_RNG` again but didn't think about it. But I have an impression that for our rand API, it would be better to recommend using a function than a global variable, this gives more freedom evolving the API without breaking change (e.g. `defaultRNG` could give a different object depending on the current thread in the future). See also #23205. But this was not the intent in this current PR to push this change, and now that it's not needed anymore, I should probably revert this bit.

Reply:
> I didn't really sea this as CamelCased, more as "initials get capitalized"

We never capitalize anything in function names, only in type names and consts.

> what would be prefered here? default_rng ? (defaultrng is un-readable).

I'd say `defaultrng`, though I think just using the const is fine; I don't see how changing the return value of the function would be any more or less breaking than changing the value of a const variable.

Reply:
> I don't see how changing the return value of the function would be any more or less breaking than changing the value of a const variable.

It's less breaking in that if `defaultRNG()` returns a different object depending on the thread (in which case `GLOBAL_RNG` may be deleted), then it won't be necessary to update rand-related functions (like `sprand`) to default to `defaultRNG()` instead of `GLOBAL_RNG`. But we are not there yet (quite some uncertainty remains regarding the future of `GLOBAL_RNG`), and needing to access `GLOBAL_RNG` directly should be quite rare anyway, so I will revert this part.